src/segy/indexing.py

```diff
@@ -251,12 +251,15 @@ def decode(self, buffer: bytearray) -> NDArray[Any]:
 
     def post_process(
         self, data: NDArray[Any]
-    ) -> NDArray[Any] | dict[str, NDArray[Any] | DataFrame]:
-        """Either return struct array or (Header) DataFrame + (Data) Array."""
-        if self.settings.use_pandas:
-            return {"header": DataFrame(data["header"]), "data": data["data"]}
+    ) -> NDArray[Any] | dict[str, NDArray[Any] | HeaderAccessor]:
+        """Return header and samples in dict.
 
-        return data
+        Data is already byte-swapped at decode so not applying it here.
+        We need to standardize the transformations in a way that's
+        applicable to trace data and headers etc.
+        """
+        header_accessor = HeaderAccessor(data=data["header"])
+        return {"header": header_accessor, "data": data["data"]}
```

Comment:
Will this return a DataFrame for the header's without applying transforms?

Reply:
that returns a header accessor; then its up to you to view it as struct array or data frame with `to_dataframe()` :-) Transforms only get applied when you ask for the data.